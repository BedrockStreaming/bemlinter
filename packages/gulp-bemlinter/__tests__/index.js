const gulp = require('gulp');
const bemlinter = require('./../src/gulp-bemlinter.js');

describe('gulp-bemlinter', () => {
  it('should lint', (done) => {
    const spy = (output) => {
      expect(output).toMatchSnapshot();
      done();
    };
    
    gulp.src('../bemlinter/__tests__/sources/*.scss')
      .pipe(bemlinter())
      .pipe(bemlinter.format(false, spy));
  });
});