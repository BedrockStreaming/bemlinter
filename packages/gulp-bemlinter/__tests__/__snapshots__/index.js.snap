exports[`gulp-bemlinter should lint 1`] = `
"  ✗ leak
[leak.scss:10] Error: \".external-block\" is incoherent with the file name.
  ✗ prefix
[prefix.scss:1] Error: \".c-prefix\" is incoherent with the file name.
[prefix.scss:26] Error: \".c-prefix__close\" is incoherent with the file name.
  ✗ syntax
[syntax.scss:21] Error: \".syntax---big\" modifier should have only 2 dashes.
[syntax.scss:26] Error: \".syntax--big--light\" should have a single modifier.
[syntax.scss:32] Error: \".syntax__Close\" should be in lower case.
[syntax.scss:40] Error: \".syntax--big__close\" represents an element of a modifier, it should be cut in 2 classes.
[syntax.scss:49] Error: \".syntax__header__logo\" should have a single depth of element.
[syntax.scss:53] Error: \".syntax___footer\" element should have only 2 underscores.
  ✓ warning
[warning.scss:21] Warning: \".external-block\" is only tolerated in this stylesheet.

FAIL: bemlinter has detected 9 errors on 3 blocks."
`;
